File: backend/app/schemas/analytics.py
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel


class MetricResponse(BaseModel):
    id: UUID
    campaign_id: UUID
    channel: str
    impressions: int
    clicks: int
    conversions: int
    ctr: float
    spend: float
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/analytics.ts
import { apiClient } from '@/lib/api-client'
import { APIResponse } from '@/types/api'

export interface CampaignMetricData {
  id: string
  campaign_id: string
  channel: string
  impressions: number
  clicks: number
  conversions: number
  ctr: number
  spend: number
  created_at: string
}

export async function fetchCampaignMetrics(campaignId: string): Promise<CampaignMetricData[]> {
  const res = await apiClient<APIResponse<CampaignMetricData[]>>(`/analytics/${campaignId}`)
  return res.data
}
